### DISCUSSION TITLE
Prepared statements don't seem to be working

### DISCUSSION THREAD

**Discussion by doconnoronca** [pgbouncer/pgbouncer#1015] [Q&A] [answered]
I have pgbouncer 1.21 installed via Gentoo. I mainly want to use it for prepared statement sharing across connections, but it doesn't seem to be working. I have enabled it with:

```
pool_mode = session
max_prepared_statements = 500
min_pool_size = 10
```
When I run it, the queries run successfully, but with "show servers" the prepared_statements column remains at 0. I using Free Pascal's sqldb library to run the queries which I'm pretty sure prepares all statements before running them.

Is there a good way to see exactly what queries at being sent to the servers so I can better understand what might be causing the problem?

**Comment by Melkij** [comment, marked as answer]
> pool_mode = session

https://www.pgbouncer.org/config.html

> When this is set to a non-zero value PgBouncer tracks protocol-level named prepared statements related commands sent by the client **in transaction and statement pooling mode**.

**Reply by JelteF** [reply]
To clarify even further, in session pooling mode prepared statements work fine, but they are not shared across connections. Session pooling intentionally works as close to regular Postgres as possible.
